Installation of 2.01

rakhmanov opened the pull request:
My android app refused to backup, it was time for update. Realized I have to revisit everything from scratch. Leading to this PR.

# Moved to it's own fork:  https://github.com/Rakhmanov/immich-in-lxc

### Steps to validate
1. Checkout this repository with the tag. Into a new folder as in the example.
`git clone --branch fix_2.01 --single-branch https://github.com/Rakhmanov/immich-in-lxc.git /home/immich/immich-in-lxc-v2` 
3. Run pre-install.sh as root
4. Run install.sh as immich user. 
When prompted - confirm removal of old version and everything under "/home/immich/app" folder by typing Y.
6. Start services: 
`systemctl start immich-web.service immich-ml.service` as root

### Debugging:
1. `systemctl status immich-web.service immich-ml.service`
2. `tail -f /var/log/immich/web.log`
3. `strace -e trace=file -f /home/immich/app/start.sh`
4. `node -p "require('sharp').versions"` to narrow down if sharp module causing the failure

### Changes
- Built on top of amazing #114
- Sharp build from source. 
- Removed dead code.
- Function calling in install and pre-install brought to single place in the bottom for ease of turning on/off.
- install is self sufficient now to generate hash from base-images - necessary for update process. Logic is copied from pre-install.
- image magic now includes raw extension.
  - Fixing raw extension bug.
  - new build dependencies are introduced and then cleaned up.
- install script triggers warning about deletion of app/*

tautomer committed:
Create auto_build_immich.yml

tautomer committed:
Update auto_build_immich.yml

Fix uid error and some wrong commands

tautomer committed:
Update auto_build_immich.yml

Use actions/setup-node@v4.3.0 instead

tautomer committed:
Update auto_build_immich.yml

Install Jellyfin ffmpeg and fix bugs

tautomer committed:
Update auto_build_immich.yml

Fix a bug that workflow exits before compiling the server

tautomer committed:
Update auto_build_immich.yml

Fix bugs in the script and optimize workflow

tautomer committed:
Update auto_build_immich.yml

Use sudo when directories under /mnt

tautomer committed:
Fix install.sh for v1.130

tautomer committed:
Fix wrong checkout in install.sh

tautomer committed:
Update auto_build_immich.yml

Add /home/runner/geodata to the artifact and release the artifact when there is a newer version of immich

tautomer committed:
Update install.sh

tautomer committed:
Update auto_build_immich.yml

Fix typos in tarball filenames

tautomer committed:
Update auto_build_immich.yml

Fix wrong needs in Release-immich-build

tautomer committed:
Change install path to /home/immich

tautomer committed:
Fix switch user bug

tautomer committed:
Fix a bug in environment variables

tautomer committed:
Fix path bug in nvm and venv

tautomer committed:
Update release tasks

tautomer committed:
Improve release description

tautomer committed:
Fix machine learning part for v.1.131

1. Add another if when copying the ML folder
2. Add an if to update systemd file for ML

tautomer committed:
Merge branch '130_131_fixes'

tautomer committed:
Merge branch '130_131_fixes'

tautomer committed:
Update installation instruction in the release info

tautomer committed:
Fix actions caching issue

tautomer committed:
Merge remote-tracking branch 'upstream/main'

tautomer committed:
Fix a bug where wrong cache is read

tautomer committed:
Fix version caching bug

tautomer committed:
Merge remote-tracking branch 'upstream/main'

tautomer committed:
Update for https://github.com/immich-app/base-images/pull/203

tautomer committed:
Update workflow with changes from upstream

tautomer committed:
Update install.sh for immich-app/base-images#203

tautomer committed:
Merge remote-tracking branch 'upstream/main'

tautomer committed:
Use GitHub repo variables

tautomer committed:
Merge remote-tracking branch 'upstream/main'

tautomer committed:
Update for 136

tautomer committed:
Fix python executable issue

tautomer committed:
fix get-cpu.sh for 136

tautomer committed:
Enable building sharp

tautomer committed:
Fix repo variable update issue

tautomer committed:
Enforce numpy<2

tautomer committed:
Fix a switching user issue

tautomer committed:
Remove intel driver; fix numpy

tautomer committed:
Refactor building workflow

tautomer committed:
final numpy fix

tautomer committed:
Revert "Refactor building workflow"

This reverts commit b0266945ac135342e47409e851af3aa5a89f8472.

tautomer committed:
Fix a missing semicolon

tautomer committed:
Update workflow for 1.138

tautomer committed:
Fix missing quotes from sed

tautomer committed:
Use a repo variable for UPLOAD_DIR

tautomer committed:
Add a workflow to pack dependencies

tautomer committed:
fix wrong user in test workflow

tautomer committed:
add dummy workflows for testing purposes

tautomer committed:
Fix install.sh for 139

tautomer committed:
Merge branch 'fix_1.139'

tautomer committed:
Refactor GitHub Actions workflows

1. Auto-build dependencies together with the server
2. Use separate jobs for different steps of building
3. Improve workflow efficiency by optimizing caching strategies
4. Improve workflow efficiency by adjusting building logics
5. Refactor for better maintainability

tautomer committed:
Use a different way for auto-trigger

tautomer committed:
Use GITHUB_TOKEN for release

tautomer committed:
Build sharp from source

tautomer committed:
Force build sharp from source

tautomer committed:
Merge building workflows back to one file

tautomer committed:
Fix unknown $ACTION for autoruns

tautomer committed:
Fix missing base-image hash

rakhmanov committed:
Version 2 install

rakhmanov commented:
I'm currently experiencing and issue: sigfault on web server side.

Can be reproduced by calling:
`node -p "require('sharp').versions"`
`Segmentation fault`

The issue is the sharp - so in the build currently uses the global libvips and then we switch it to the built in libvips. However what we want is to build sharp against global libvips and we should not switch around anything. Tomorrow I will build sharp in the pre-install step, and then carry it over in to the install.

damianpeterson commented:
I've just tried installing this from scratch. Pulled `https://github.com/Rakhmanov/immich-in-lxc/tree/fix_2.01` into Ubuntu 24.04 on Proxmox 8.2.2. Followed the installation instructions in the homepage readme. While running the `./pre-install.sh` as root it failed with:
```
CREATE EXTENSION
+ change_permission
+ chmod 666 '/home/immich/immich-in-lxc/base-images/server/sources/*.json'
chmod: cannot access '/home/immich/immich-in-lxc/base-images/server/sources/*.json': No such file or directory
```
I've not looked any further into it but happy to continue further testing if you need.

[edit: I removed the `change_permission` function and call from `pre-install.sh` and have been able to continue with the installation. I'm now at the point where I'm in a loop of:

```
Started immich-web.service - immich web server.
immich-web.service: Main process exited, code=killed, status=11/SEGV
immich-web.service: Failed with result 'signal'.
...
```

which is I presume where you're up to with your segmentation fault?]

rakhmanov commented:
Hey, thanks for running it. 

I believe I have a solve for the sigfault.
And its basically an npm call for a rebuild of the sharp module inside of the app folder. 

So what was happening we would build it originally but since pnmp does clever symlinking "deploy" does not properly copy binaries into the target folder (app) from (source) where we have built against the globally installed vips. Therefore its just easier to rebuild and it all works. This has small downside, is that we cant really delete all the build dependencies from the pre-install - this has been commented out anyway so I take it. We could move the cleanup to the post-install step.

~~I will push the changes, as soon as my thumb rebuild completes. So I can test it back to back again.~~ - Done

rakhmanov committed:
Rebuildig sharp, and returning base image pull to pre-install.sh

damianpeterson commented:
More errors, I'm afraid.

I started from a clean instance and during the `install.sh` step got the following:

```
Updating existing base-images repo...
fatal: detected dubious ownership in repository at '/home/immich/immich-in-lxc/base-images'
To add an exception for this directory, call:

        git config --global --add safe.directory /home/immich/immich-in-lxc/base-images
```

I added that as a safe directory and re-ran `install.sh` but this time it failed with:

```
Updating existing base-images repo...
error: cannot open '.git/FETCH_HEAD': Permission denied
```

rakhmanov commented:
This is because the base images were checked out as a root. And we are running the install.sh as a immich. 

3 Options to resolve: 
- Make immich an owner `chown immich:immich -R /home/immich/immich-in-lxc/base-images`. 
- delete folder and let it be recreated.
- or checkout the branch into a new folder and let it be recreated.

~~I will adjust the script to checkout as non root - which is correct solution.~~ Done.

rakhmanov committed:
Introduced helper script for reuse, correct checkout permissions

rakhmanov commented:
Pushed new changes that deal with permissions, we now are checking out as immich user by default. This will ensure we will have no permissions issues going forward when install script needs to do anything. 

I've commented out `generate_build_lock` to see if immich indeed would complain (as it implied in the comment next to this function call). And I have not seen it complain about the signature not being there. If no one else has issues we could delete it all together - simplifying the logic.

damianpeterson commented:
That worked flawlessly. Nice one.

Elhorian commented:
I'm trying to use this one with Debian 12 VM and it seems to have issues with ML (openvino) as it uses 1.18 which is waaay older than current versions. In my previous issues someone had problems also with this and ended up downgrading to Python 3.11, which is not ideal. Is there any other thing that can be done to fix it instead?

rakhmanov commented:
> I'm trying to use this one with Debian 12 VM and it seems to have issues with ML (openvino) as it uses 1.18 which is waaay older than current versions. In my previous issues someone had problems also with this and ended up downgrading to Python 3.11, which is not ideal. Is there any other thing that can be done to fix it instead?

Hey, this is unrelated to this PR. I don't use Debian neither openvino on my system, but someone else might. Please create an Issue describing in detail what you are experiencing.

damianpeterson commented:
Just a follow-up to say this has been working smoothly since the initial install. 

I migrated 21,000 (65GiB) photos across and re-ran ML smart search and facial recognition from scratch. Just needed to change the default URL for ML to localhost.

Ubuntu 24.04 on Proxmox 8.2.2 in an LXC container with 10GiB RAM, 30GiB root disk (+ separate mount for photos/uploads).

Thanks for your good work, @Rakhmanov

tomamplius commented:
I have test on my debian install

~~~
thomas@ubuntu:~#  ssh promox-pve3
root@pve3:~# pct enter 240
root@immich:~# su - immich -s /bin/bash
immich@immich:~$ git clone --branch fix_2.01 --single-branch https://github.com/Rakhmanov/immich-in-lxc.git immich-in-lxc-v2/
Cloning into 'immich-in-lxc-v2'...
remote: Enumerating objects: 1410, done.
remote: Counting objects: 100% (468/468), done.
remote: Compressing objects: 100% (79/79), done.
remote: Total 1410 (delta 439), reused 395 (delta 389), pack-reused 942 (from 3)
Receiving objects: 100% (1410/1410), 279.07 KiB | 3.58 MiB/s, done.
Resolving deltas: 100% (864/864), done.
immich@immich:~$ exit
logout
root@immich:~# cd ~immich/immich-in-lxc-v2/
root@immich:/var/lib/immich/immich-in-lxc-v2# ./pre-install.sh 
+ set_common_variables
+ set -a
+ SCRIPT_DIR=/var/lib/immich/immich-in-lxc-v2
+ REPO_URL=https://github.com/immich-app/base-images
+ APP_REPO_URL=https://github.com/immich-app/immich
+ BASE_IMG_REPO_DIR=/var/lib/immich/immich-in-lxc-v2/base-images
+ SOURCE_DIR=/var/lib/immich/immich-in-lxc-v2/image-source
+ LD_LIBRARY_PATH=/usr/local/lib
+ LD_RUN_PATH=/usr/local/lib
+ set_user_to_run
+ [[ -n '' ]]
+ id immich
+ USER_TO_RUN=immich
+ export USER_TO_RUN
+ set +a
+ safe_git_checkout https://github.com/immich-app/base-images /var/lib/immich/immich-in-lxc-v2/base-images main
+ set_user_to_run
+ [[ -n '' ]]
+ id immich
+ USER_TO_RUN=immich
+ export USER_TO_RUN
+ local repo_url=https://github.com/immich-app/base-images
+ local target_dir=/var/lib/immich/immich-in-lxc-v2/base-images
+ local ref=main
++ id -un
+ [[ root == \i\m\m\i\c\h ]]
++ id -un
+ [[ root != \r\o\o\t ]]
+ echo 'Switching to immich for git operations...'
Switching to immich for git operations...
+ sudo -u immich bash -c '
        source '\''./helpers.sh'\''
        git_checkout_repo '\''https://github.com/immich-app/base-images'\'' '\''/var/lib/immich/immich-in-lxc-v2/base-images'\'' '\''main'\''
    '
./helpers.sh: line 64: sudo: command not found

rakhmanov commented:
> I have test on my debian install
> 
> ```
> thomas@ubuntu:~#  ssh promox-pve3
> root@pve3:~# pct enter 240
> root@immich:~# su - immich -s /bin/bash
> immich@immich:~$ git clone --branch fix_2.01 --single-branch https://github.com/Rakhmanov/immich-in-lxc.git immich-in-lxc-v2/
> Cloning into 'immich-in-lxc-v2'...
> remote: Enumerating objects: 1410, done.
> remote: Counting objects: 100% (468/468), done.
> remote: Compressing objects: 100% (79/79), done.
> remote: Total 1410 (delta 439), reused 395 (delta 389), pack-reused 942 (from 3)
> Receiving objects: 100% (1410/1410), 279.07 KiB | 3.58 MiB/s, done.
> Resolving deltas: 100% (864/864), done.
> immich@immich:~$ exit
> logout
> root@immich:~# cd ~immich/immich-in-lxc-v2/
> root@immich:/var/lib/immich/immich-in-lxc-v2# ./pre-install.sh 
> + set_common_variables
> + set -a
> + SCRIPT_DIR=/var/lib/immich/immich-in-lxc-v2
> + REPO_URL=https://github.com/immich-app/base-images
> + APP_REPO_URL=https://github.com/immich-app/immich
> + BASE_IMG_REPO_DIR=/var/lib/immich/immich-in-lxc-v2/base-images
> + SOURCE_DIR=/var/lib/immich/immich-in-lxc-v2/image-source
> + LD_LIBRARY_PATH=/usr/local/lib
> + LD_RUN_PATH=/usr/local/lib
> + set_user_to_run
> + [[ -n '' ]]
> + id immich
> + USER_TO_RUN=immich
> + export USER_TO_RUN
> + set +a
> + safe_git_checkout https://github.com/immich-app/base-images /var/lib/immich/immich-in-lxc-v2/base-images main
> + set_user_to_run
> + [[ -n '' ]]
> + id immich
> + USER_TO_RUN=immich
> + export USER_TO_RUN
> + local repo_url=https://github.com/immich-app/base-images
> + local target_dir=/var/lib/immich/immich-in-lxc-v2/base-images
> + local ref=main
> ++ id -un
> + [[ root == \i\m\m\i\c\h ]]
> ++ id -un
> + [[ root != \r\o\o\t ]]
> + echo 'Switching to immich for git operations...'
> Switching to immich for git operations...
> + sudo -u immich bash -c '
>         source '\''./helpers.sh'\''
>         git_checkout_repo '\''https://github.com/immich-app/base-images'\'' '\''/var/lib/immich/immich-in-lxc-v2/base-images'\'' '\''main'\''
>     '
> ./helpers.sh: line 64: sudo: command not found
> ```

Please try installing sudo. Its relies on it to change user to I'm much when doing the checkout. If this works I can add sudo as dependency.

tomamplius commented:
sudo is command to give right to normal user. Why use sudo from root to change user to immich?
Upgrade in progress after sudo install

su - immich -s /bin/bash  'xxxxxxxxxxxxxxxxxxx'  work well without dependency

tomamplius commented:
after many issue on upgrade i have finally remove /var/lib/immich/source before launch install.sh

All is done on debian 13 install with v1.135.3 and upgrade on 2.0.1 with this script

rakhmanov commented:
Hey if you have any suggestions pls feel free to pr them into my repo/branch. I will be happy to merge it. Checking out into a clean folder or removing existing code is part of the instructions.

tomamplius commented:
sorry when i meed issue on upgrade, it because during compilation some files are update then git pull not work because some file are change and same type of issue on poetry.
I have remove all git repository download in /source during the first install to start on clean immich source code
Except sudo requirement add all the script is OK

horstt commented:
Just tried it on my relatively fresh 1.143.1 version. The pre-install went smoothly, but the install.sh stopped abruptly, no serious errors, it just said geodata already present and stopped the upgrade:
`  - Installing uvicorn (0.38.0)
Installing dependencies from lock file

Package operations: 6 installs, 0 updates, 0 removals

  - Installing humanfriendly (10.0)
  - Installing mpmath (1.3.0)
  - Installing coloredlogs (15.0.1)
  - Installing flatbuffers (25.9.23)
  - Installing sympy (1.14.0)
  - Installing onnxruntime (1.23.1)
Removing source with name langsam.
geodata exists, skip downloading
immich@immich:~/immich-in-lxc-v2$`


 Rebooted the container and it worked, but it was still at 1.143.1. Now I deleted the contents of the /home/immich/source folder and restarted the upgrade by running the sudo ./pre-install.sh command, which worked fine, but running the ./install.sh as immich user results in this error now:
 ERR_PNPM_NO_PKG_MANIFEST  No package.json found in /home/immich/source

I guess I deleted a bit too much from /home/immich/source :-p 

Will restore the backup and retry, but do you have any advice for me, to upgrade a working version?

=========================
Update: restored backup, removed geodata and retried. Now it installs geodata but it stops after it has succeeded.

rakhmanov commented:
The question did it actually have the package.json file, or not. It could be just permission issue. The files in the source folder should be immich:immich owner, that way install script sees them. What are you seeing now, what is it stopping on, which OS.

horstt commented:
> The question did it actually have the package.json file, or not. It could be just permission issue. The files in the source folder should be immich:immich owner, that way install script sees them. What are you seeing now, what is it stopping on, which OS.

The folder contained the package file, until I deleted it ;-) Now, after the restore, it again contains the package.json file. Like I said, the install script seems to stop running without error when it detects I have a working installation, because that sources folder is filled with files. But when I empty the folder beforehand, as @tomamplius described, it cannot find a package.json file, because I wiped it. Should the pre-install script will this folder? 

My installation is based on Ubuntu 24.04. It's a relatively fresh container, since I had upgrade issues from 1.135.x to 1.14xxx. At that time, I made a new container, installed the current live version on Github, and used the 1.139 upgrade that was mentioned earlier (https://github.com/loeeeee/immich-in-lxc/pull/114). I managed to upgrade that up to 1.143.1.

rakhmanov commented:
You should be able to delete source folder completely and have it fully be rebuilt in the install.sh script. Thats what its for and supposed to do in clean env.  
So now when you delete the source folder, run everything again, and check the content of the source folder (ls -lha) when you are getting the failure.

Ensure to see owner of the folder and files.  It should contain everything you see under the tag `https://github.com/immich-app/immich/tree/v2.0.1`.
If that is not there than it means that install.sh step `safe_git_checkout "$REPO_URL" "$INSTALL_DIR_src" "$REPO_TAG"` Is not doing what it supposed to, which is git clone the repo in the source folder. If that the case try manually running (from within /home/immich) `git clone --branch v2.0.1 https://github.com/immich-app/immich.git  source`. 

Copy last few lines fom the execution history to narrow down what was attempted.

horstt commented:
Okay that was stupid of me, before I just attempted to delete the contents of the source folder, and left behind the hidden .git folder. 

Now I deleted the whole source folder and the installation went better. But... Didn't succeed. I still got stuck at the point where the script notices I already have a functioning installation, because my geodata folder has contents. There, the script tells me I have geodata present and the download of geodata is cancelled, at which point the script stops. 

These are the last lines from the execution history:

```
Using cached h11-0.16.0-py3-none-any.whl (37 kB)
Using cached sniffio-1.3.1-py3-none-any.whl (10 kB)
Using cached typing_extensions-4.15.0-py3-none-any.whl (44 kB)
Using cached pycparser-2.23-py3-none-any.whl (118 kB)
Installing collected packages: trove-classifiers, fastjsonschema, distlib, zstandard, urllib3, typing_extensions, tomlkit, sniffio, shellingham, rapidfuzz, pyproject-hooks, pycparser, poetry-core, platformdirs, pkginfo, pbs-installer, packaging, msgpack, more-itertools, jeepney, jaraco.context, installer, idna, h11, filelock, crashtest, charset_normalizer, certifi, virtualenv, requests, jaraco.functools, jaraco.classes, httpcore, findpython, dulwich, cleo, cffi, build, anyio, requests-toolbelt, httpx, cryptography, cachecontrol, SecretStorage, keyring, poetry
Successfully installed SecretStorage-3.4.0 anyio-4.11.0 build-1.3.0 cachecontrol-0.14.3 certifi-2025.10.5 cffi-2.0.0 charset_normalizer-3.4.4 cleo-2.1.0 crashtest-0.4.1 cryptography-46.0.3 distlib-0.4.0 dulwich-0.24.6 fastjsonschema-2.21.2 filelock-3.20.0 findpython-0.7.0 h11-0.16.0 httpcore-1.0.9 httpx-0.28.1 idna-3.11 installer-0.7.0 jaraco.classes-3.4.0 jaraco.context-6.0.1 jaraco.functools-4.3.0 jeepney-0.9.0 keyring-25.6.0 more-itertools-10.8.0 msgpack-1.1.2 packaging-25.0 pbs-installer-2025.10.14 pkginfo-1.12.1.2 platformdirs-4.5.0 poetry-2.2.1 poetry-core-2.2.1 pycparser-2.23 pyproject-hooks-1.2.0 rapidfuzz-3.14.1 requests-2.32.5 requests-toolbelt-1.0.0 shellingham-1.5.4 sniffio-1.3.1 tomlkit-0.13.3 trove-classifiers-2025.9.11.17 typing_extensions-4.15.0 urllib3-2.5.0 virtualenv-20.35.3 zstandard-0.25.0
Adding source with name langsam.
Updating dependencies
Resolving dependencies... (29.6s)

Package operations: 105 installs, 0 updates, 0 removals

  - Installing wsproto (1.2.0)
  - Installing markupsafe (3.0.3)
  - Installing numpy (1.26.4)
  - Installing setuptools (80.9.0)
  - Installing simple-websocket (1.1.0)
  - Installing annotated-types (0.7.0)
  - Installing bidict (0.23.1)
  - Installing blinker (1.9.0)
  - Installing click (8.3.0)
  - Installing greenlet (3.2.4)
  - Installing itsdangerous (2.2.0)
  - Installing jinja2 (3.1.6)
  - Installing opencv-python-headless (4.11.0.86)
  - Installing pillow (10.4.0)
  - Installing pydantic-core (2.41.4)
  - Installing python-engineio (4.12.3)
  - Installing simsimd (6.5.3)
  - Installing six (1.17.0)
  - Installing stringzilla (4.2.1)
  - Installing typing-inspection (0.4.2)
  - Installing websocket-client (1.9.0)
  - Installing werkzeug (3.1.3)
  - Installing zope-event (6.0)
  - Installing zope-interface (8.0.1)
  - Installing albucore (0.0.24)
  - Installing brotli (1.1.0)
  - Installing configargparse (1.7.1)
  - Installing contourpy (1.3.3)
  - Installing cycler (0.12.1)
  - Installing flask (3.1.2)
  - Installing fonttools (4.60.1)
  - Installing fsspec (2025.9.0)
  - Installing gevent (25.5.1)
  - Installing hf-xet (1.1.10)
  - Installing imageio (2.37.0)
  - Installing iniconfig (2.3.0)
  - Installing joblib (1.5.2)
  - Installing kiwisolver (1.4.9)
  - Installing lazy-loader (0.4)
  - Installing mdurl (0.1.2)
  - Installing ml-dtypes (0.5.3)
  - Installing networkx (3.5)
  - Installing pluggy (1.6.0)
  - Installing protobuf (6.33.0)
  - Installing pydantic (2.12.3)
  - Installing pygments (2.19.2)
  - Installing pyparsing (3.2.5)
  - Installing python-dateutil (2.9.0.post0)
  - Installing python-socketio (5.14.2)
  - Installing pyyaml (6.0.3)
  - Installing scipy (1.16.2)
  - Installing threadpoolctl (3.6.0)
  - Installing tifffile (2025.10.16)
  - Installing tqdm (4.67.1)
  - Installing wcwidth (0.2.14)
  - Installing albumentations (2.0.8)
  - Installing coverage (7.11.0)
  - Installing cython (3.1.5)
  - Installing easydict (1.13)
  - Installing flask-cors (6.0.1)
  - Installing flask-login (0.6.3)
  - Installing geventhttpclient (2.3.4)
  - Installing httptools (0.7.1)
  - Installing huggingface-hub (0.35.3)
  - Installing locust-cloud (1.27.6)
  - Installing markdown-it-py (4.0.0)
  - Installing matplotlib (3.10.7)
  - Installing mypy-extensions (1.1.0)
  - Installing onnx (1.19.1)
  - Installing pathspec (0.12.1)
  - Installing prettytable (3.16.0)
  - Installing psutil (7.1.1)
  - Installing pytest (8.4.2)
  - Installing python-dotenv (1.1.1)
  - Installing pytokens (0.2.0)
  - Installing pyzmq (27.1.0)
  - Installing scikit-image (0.25.2)
  - Installing scikit-learn (1.7.2)
  - Installing starlette (0.48.0)
  - Installing uvloop (0.22.1)
  - Installing watchfiles (1.1.1)
  - Installing websockets (15.0.1)
  - Installing aiocache (0.12.3)
  - Installing black (25.9.0)
  - Installing fastapi (0.119.1)
  - Installing ftfy (6.3.1)
  - Installing gunicorn (23.0.0)
  - Installing insightface (0.7.3)
  - Installing locust (2.41.6)
  - Installing mypy (1.18.2)
  - Installing orjson (3.11.3)
  - Installing pydantic-settings (2.11.0)
  - Installing pytest-asyncio (1.2.0)
  - Installing pytest-cov (7.0.0)
  - Installing pytest-mock (3.15.1)
  - Installing python-multipart (0.0.20)
  - Installing rich (14.2.0)
  - Installing ruff (0.14.1)
  - Installing tokenizers (0.22.1)
  - Installing types-pyyaml (6.0.12.20250915)
  - Installing types-requests (2.32.4.20250913)
  - Installing types-setuptools (80.9.0.20250822)
  - Installing types-simplejson (3.20.0.20250822)
  - Installing types-ujson (5.10.0.20250822)
  - Installing uvicorn (0.38.0)
Installing dependencies from lock file

Package operations: 6 installs, 0 updates, 0 removals

  - Installing humanfriendly (10.0)
  - Installing mpmath (1.3.0)
  - Installing coloredlogs (15.0.1)
  - Installing flatbuffers (25.9.23)
  - Installing sympy (1.14.0)
  - Installing onnxruntime (1.23.1)
Removing source with name langsam.
geodata exists, skip downloading
immich@immich:~/immich-in-lxc-v2$
```

Update:
I was wondering what'd happen if I deleted the contents from the geodata folder. After deleting the geodata the script does download the geodata, but halts right after that download has succeeded:
```
 11800K .......... .......... .......... .......... .......... 96% 35.0M 0s
 11850K .......... .......... .......... .......... .......... 97% 34.3M 0s
 11900K .......... .......... .......... .......... .......... 97% 33.6M 0s
 11950K .......... .......... .......... .......... .......... 98% 33.7M 0s
 12000K .......... .......... .......... .......... .......... 98% 23.9M 0s
 12050K .......... .......... .......... .......... .......... 98% 31.3M 0s
 12100K .......... .......... .......... .......... .......... 99% 33.4M 0s
 12150K .......... .......... .......... .......... .......... 99% 36.6M 0s
 12200K .......... .......... .......... ..                   100% 56.3M=0.6s

2025-10-21 20:56:00 (19.5 MB/s) - ‘cities500.zip’ saved [12526362/12526362]

Archive:  cities500.zip
  inflating: cities500.txt
immich@immich:~/immich-in-lxc-v2$
```

rakhmanov commented:
So it stops after the geo step, and does not seem to produce error. You could check with `echo $?`. 0 is good 1 is bad 

 there are 2 more steps left which it seems not to do anything with:

create_custom_start_script
create_runtime_env_file

Could they be commented out by any chance?

horstt commented:
> So it stops after the geo step, and does not seem to produce error. You could check with `echo $?`. 0 is good 1 is bad
> 
> there are 2 more steps left which it seems not to do anything with:
> 
> create_custom_start_script create_runtime_env_file
> 
> Could they be commented out by any chance?

No, I did not edit the install script before. 

Now I did, as I assume I need to insert the commando echo $? after the geodata line?

These are the last lines in the install script now:

```
# generate_build_lock <- I dont know if we stil need it I havent had immich complaining
install_immich_machine_learning
replace_usr_src
setup_upload_folder
download_geonames
echo $?
create_custom_start_script
create_runtime_env_file
```
Ran the script again, the following are the last lines of the result:
```
Writing lock file
Installing dependencies from lock file

Package operations: 6 installs, 0 updates, 0 removals

  - Installing humanfriendly (10.0)
  - Installing mpmath (1.3.0)
  - Installing coloredlogs (15.0.1)
  - Installing flatbuffers (25.9.23)
  - Installing sympy (1.14.0)
  - Installing onnxruntime (1.23.2)
Removing source with name langsam.
geodata exists, skip downloading
0
immich@immich:~/immich-in-lxc-v2$ echo $?
0
```

horstt commented:
Ehm... Maybe a funny twist, when I was looking through my database backups, I noticed my last fewe backups were called something along the range of:
immich-db-backup-20251026T020000-v2.1.0-pg17.6.sql

It seems the upgrade went through, at least mostly, but the UI still thinks it's at 1.143.1. Any clue what's up with that?

horstt commented:
Okay, managed to install 2.0.1 on an empty container, using your script. Restored database and everything so I have a working version 2.0.1 now, with all my photos. Funny thing is, when I upgrade this install from 2.0.1 to 2.1.0, the upgrade works, but also doesn't upgrade the version number in the UI...

RudolfRendier commented:
Anyone has advice on how to get from 123 to the latest?

I need to perform this typeORM upgrade:
https://docs.immich.app/errors/

What commit of this repo/what immich version can I use as intermediate step?

horstt commented:
> Anyone has advice on how to get from 123 to the latest?

First and foremost: MAKE BACKUPS. Then I'd upgrade in a few steps... You can try an upgrade to the version the current master is based on: 1.135.3. See if your database upgrades correctly. Then you can try the upgrade to 1.139.2 that's linked above. Or go straight to version 2.0.1 that @Rakhmanov describes here...
 
Mind you, I'm not stable at the moment, and I've needed to make a clean new container twice, migrating my data.

rakhmanov commented:
> Mind you, I'm not stable at the moment, and I've needed to make a clean new container twice, migrating my data.

Anything related to PR?

RudolfRendier commented:
Upgrade to 2.0.1 was successful.
Performed a few intermediate steps 131, 133, 135 and 201.
Every checkout ran the pre-install as root, followed by install as immich.
For 131 I had to checkout an older version of base-images and prevent the git_clone step from undoing that.
Had to fix my Nginx Proxy for 133 and up and had to manually download vectorchord for postgres 16.

Thanks for your efforts.

horstt commented:
> > Mind you, I'm not stable at the moment, and I've needed to make a clean new container twice, migrating my data.
> 
> Anything related to PR?

Well, initial installation of 2.0.1 in a clean container with your script works. However, every upgrade afterwards fails, in the way I described. The script halts after doing the geolocation stuff, without error or warning, and without actually upgrading the web-side of the immich server. I have no clue how to fix,  this issue and started on the path of the arter97 immich-native. But in that path I'm also getting myself tangled up with stuff not working, since that build doesn't support heic images and my wife uses her iPhone to make pictures...  

https://github.com/arter97/immich-native/issues/9

rakhmanov commented:
* Fixed update - force checkout the version specified in .env
* Added install complete message.

What was happening is somewhere during installation of immich itself it was adding 3 lines in a source folder file, this made git checkout fail and stay on current branch due to local changes overwrite prevention. Now it will force checkout overwriting the changes in source file. 

@horstt  Heic file support - I downloaded HEIC file from Iphone 12 pro and uploaded into immich and it created preview no problem. Send me a file so I can test. This branch is actually fixes a lot of raw file support. Which I use. 

Update:
~~Crashed - due to OCR issue. Checking if setting lower version solves it.~~
Pushed update fixing version to the one that not causing the crash thanks to @bluray93 for the tip.

rakhmanov committed:
Upgrade fixes

rakhmanov committed:
Merge branch 'main' into fix_2.01

rakhmanov commented:
Hey everyone, I'm closing this in favor of continuing development in my own fork:
https://github.com/Rakhmanov/immich-in-lxc
